### PR TITLE
OCPBUGS-42425: Update base image of scaffolded Dockerfile for helm-operator to use rhel9 repository

### PR DESCRIPTION
**Pull request by arkadeepsen**
**Description of the change:**
Update base image of scaffolded Dockerfile for helm-operator to use rhel9 repository

**Motivation for the change:**
The helm-operator image is using rhel9 base image from 4.16 onwards and discontinued publishing rhel8 based image. The repository also has changed from `registry.redhat.io/openshift4/ose-helm-operator` to `registry.redhat.io/openshift4/ose-helm-rhel9-operator`. This PR fixes the scaffolded Dockerfile for helm-operator to use the correct image.

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [ ] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Review by KeenonLee — COMMENTED** [review]
/label qe-approved
Thanks~

File: internal/plugins/openshift/v1/init_test.go [hunk 1]
```diff
@@ -79,9 +79,9 @@ FROM registry.redhat.io/openshift4/ose-ansible-operator:v` + ocpProductVersion +
 FROM quay.io/operator-framework/ansible:latest
 FROM foo/ansible-operator:latest
 
-FROM registry.redhat.io/openshift4/ose-helm-operator:v` + ocpProductVersion + `
-FROM registry.redhat.io/openshift4/ose-helm-operator:v` + ocpProductVersion + `
-FROM registry.redhat.io/openshift4/ose-helm-operator:v` + ocpProductVersion + `
+FROM registry.redhat.io/openshift4/ose-helm-rhel9-operator:v` + ocpProductVersion + `
+FROM registry.redhat.io/openshift4/ose-helm-rhel9-operator:v` + ocpProductVersion + `
```
Comment (TrilokGeer) [inline internal/plugins/openshift/v1/init_test.go:83]: Is there any PR reference to the repo path change to include rhel9? Curious to understand the decision to understand the requirement.

Comment (arkadeepsen) [reply internal/plugins/openshift/v1/init_test.go:83]: @joepvd Can probably provide that context here.

Comment (joepvd) [reply internal/plugins/openshift/v1/init_test.go:83]: For better or worse, it has been a requirement that images on the container catalog have their rhel version in their name. Changing rhel-ness means changing pullspec. (This apparently was not as stictly enforced while images were moving to rhel8)

**Pull request merged**